sharktank/sharktank/types/tensors.py

```diff
@@ -1098,17 +1116,28 @@ def serialized_name(cls) -> str:
         return cls.__name__
 
     def add_to_archive(self, builder: ShardedArchiveBuilder) -> InferenceTensorMetadata:
-        for i, pt in enumerate(self._shards):
-            builder.for_rank(i).add_tensor(pt.name, pt._data)
+
         extra_properties = {
-            "shard_count": len(self._shards),
             "shape": list(self.shape),
+            "shard_count": len(self._shards),
+            "shard_type": self._shards[0].serialized_name(),
+            "extra_properties_shards": {},
```

Comment:
`extra_properties_shards` seems too long as we are already in the context of extra properties.
Maybe just `shards` or `shard_properties` is OK.